### PR TITLE
16 prepoulate the roles

### PR DESCRIPTION
**Pull request by longshuicy**
Update the enum type of roles to allow swagger to generate a dropdown based on @Rashmil-1999's suggestion
image

----
- This PR implement the same baseCRUD pattern for roles and users; adjust the foreign key logic of role-user mapping
- Maybe we can pre-populating the roles using a script once the service is up?

To test
- start the service following the readme
- go to `http://localhost:8000/docs`
- create user 
image
- create role
image
- attach role to user
image
- feel free to test other REST endpoints

### CODE REVIEW

File: app/crud/UserCRUD.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from app.models import User, Role
+from sqlalchemy.orm import Session
+
+
+class UserCRUD(
+    CRUDBase[User, UserCreate, UserUpdate]
+):
+
+    def get_user_by_email(self, db: Session, email: str) -> User:
+        return db.query(User).filter(User.email == email).first()
```
Comment (ywkim312) [inline app/crud/UserCRUD.py:14]: what if there are two emails in the db? should we check if there are multiple results and prompt warning or error? This is not just for this one. Other queries with .first() in next methods as well.

**Comment by navarroc** [comment]
How do we prevent users from creating new roles? It seems like creating a new role should be an admin privilege. Maybe we punt that for now, but we need to limit who can create roles. Can we remove the method for creating roles and add a script to this repository that we can use to populate the roles we want? Just a thought.

**Comment by longshuicy** [comment]
> How do we prevent users from creating new roles? It seems like creating a new role should be an admin privilege. Maybe we punt that for now, but we need to limit who can create roles. Can we remove the method for creating roles and add a script to this repository that we can use to populate the roles we want? Just a thought.

Let me try if i can use the x-auth-usergroup to restrict only ncsa developers can create group.

**Review by Rashmil-1999 — APPROVED** [review]

File: app/routers/roles.py [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
-    roles = crud.get_roles(db, skip=skip, limit=limit)
+    roles = crud.roleCRUD.get_multi(db, skip=skip, limit=limit)
     return roles
+
+
+@router.post("", response_model=schemas.Role)
+def create_roles(request:Request, name: RoleEnum, description: str, db: Session = Depends(get_db)):
+    # TODO wrap it as the dependency in the future
+    user_groups = request.headers.get('x-auth-usergroup')
+    if user_groups is not None:
```
[the remark this answers is not in the document]
Comment (longshuicy) [reply app/routers/roles.py:24]: Yeah. Not set the token but set the header, similar idea. 
Anyway that's just something nice to have but not super necessary

[the remark this answers is not in the document]
Comment (Rashmil-1999) [reply app/routers/roles.py:24]: Screenshot 2022-12-07 at 1 58 41 PM

Did you mean this?

[the remark this answers is not in the document]
Comment (longshuicy) [reply app/routers/roles.py:24]: Oh you got it work?!

**Pull request merged**